### PR TITLE
Ignore thumb bit when checking for repeated frames

### LINKED ISSUE #46: stack trace infinite loop
I'm just getting started using probe-run, testing a threading crate I'm building.

So (on Cortex-M), the thread stacks are populated with a frame like an exception would do, ```[r0-r4, r12, LR, PC, APSR]```.

LR is set to the location of a `cleanup()`, which does some house-keeping, then triggers the scheduler to switch away, then does a ```loop {}```, which is practically unreachable.

Now when I run something with probe-run, the resulting stack trace endlessly repeats in cleanup():

```
[...]
    Finished dev [optimized + debuginfo] target(s) in 0.01s              
     Running `probe-run --chip nRF52840_xxAA /home/kaspar/src/own/rust/riot_core/target/thumbv7em-none-eabi/debug/examples/bench_lock`             
flashing program ..                                                      
^[[CDONE                                                                 
resetting device                                                         
stack backtrace:                                                         
   0: 0x00001752 - <unknown>                                             
   1: 0x000016b6 - cortex_m_semihosting::hio::hstdout
   2: 0x00001650 - cortex_m_semihosting::export::hstdout_fmt::{{closure}}                                                                          
   3: 0x00001558 - cortex_m::interrupt::free                             
   4: 0x00001638 - cortex_m_semihosting::export::hstdout_fmt
   5: 0x000007b6 - user_main                                             
   6: 0x0000112a - riot_core::main_trampoline                            
   7: 0x00000d20 - riot_core::thread::cleanup                                                                                                      
   8: 0x00000d20 - riot_core::thread::cleanup                            
   9: 0x00000d20 - riot_core::thread::cleanup       
  10: 0x00000d20 - riot_core::thread::cleanup                            
  11: 0x00000d20 - riot_core::thread::cleanup            
  12: 0x00000d20 - riot_core::thread::cleanup
  13: 0x00000d20 - riot_core::thread::cleanup
[...]
```

The lines ```<N>: 0x00000d20 - riot_core::thread::cleanup``` repeat extremely fast, and the process cannot be cancelled with CTRL-C.

While I might have messed up something on the stack to make it invalid and look like this to probe-run, I'd expect it to stop after maybe the second iteration, letting the user know that this would repeat.

(I'm assuming that probe-run's rtt support uses a different BKPT scheme than traditional semihosting, and that's why this exits on the first ```hprintln!()```, but that is a unrelated to this issue.)

This is on probe-run 0.1.3 installed today via "cargo install probe-run".

### PR DESCRIPTION
**Pull request by jonas-schievink**
Hopefully fixes #46

### CODE REVIEW

**Comment by kaspar030** [comment]
This fixes it:

```
    Finished dev [optimized + debuginfo] target(s) in 2.59s
     Running `probe-run --chip nRF52840_xxAA target/thumbv7em-none-eabi/debug/app`
flashing program ..
DONE
resetting device
RTT logs not available; blocking until the device halts..
stack backtrace:
   0: 0x00000946 - <unknown>
   1: 0x000008aa - cortex_m_semihosting::hio::hstdout
   2: 0x000007ec - cortex_m_semihosting::export::hstdout_str::{{closure}}
   3: 0x0000071e - cortex_m::interrupt::free
   4: 0x000007d8 - cortex_m_semihosting::export::hstdout_str
   5: 0x0000045a - user_main
   6: 0x00001296 - riot_core::main_trampoline
   7: 0x00000fb2 - riot_core::thread::cleanup
error: the stack appears to be corrupted beyond this point
```

File: src/main.rs [hunk 1]
```diff
@@ -581,7 +581,7 @@ fn backtrace(
             break;
         }
 
-        if !cfa_changed && lr == pc {
+        if !cfa_changed && lr & !1 == pc & !1 {
```
Comment (kaspar030) [inline src/main.rs:584]: personally I'd expect a comment on thy the lsb is ignored. :)

Comment (japaric) [inline src/main.rs:584]: can we move the `1` constant into a `const THUMB_BIT` or similar?

**Review by japaric — APPROVED** [review]
Looks good to me but let's add a `const`

**Pull request merged**